.github/workflows/e2e_targeted.yaml

```diff
@@ -42,197 +41,53 @@ jobs:
-    strategy:
-      fail-fast: false
-      matrix:
-        test: ${{ fromJSON(needs.setup-e2e-tests.outputs.test_names) }}
-    steps:
-
-    - name: Download built image artifact
-      if: ${{ github.event.inputs.controller-image == 'kong/kubernetes-ingress-controller:ci' }}
-      uses: actions/download-artifact@v3
-      with:
-        name: kic-image
-        path: /tmp
-
-    - name: Load built image
-      if: ${{ github.event.inputs.controller-image == 'kong/kubernetes-ingress-controller:ci' }}
-      run: |
-        docker load --input /tmp/image.tar
-        docker image ls -a
-
-    - name: checkout repository
-      uses: actions/checkout@v3
-      with:
-        fetch-depth: 0
-
-    - name: setup golang
-      uses: actions/setup-go@v4
-      with:
-        go-version: '^1.20'
-
-    - uses: Kong/kong-license@master
-      id: license
-      with:
-        password: ${{ secrets.PULP_PASSWORD }}
-
-    # The only reason we need those 2 is the fact that TEST_KONG_CONTROLLER_IMAGE_LOAD
-    # is ultimately passed down to ktf's loadimage.Addon which uses "kind load docker-image ..."
-    # which fails when an image is not available on the cluster.
-    # Hence the non local version (where the image is pulled from an external registry)
-    # doesn't define TEST_KONG_CONTROLLER_IMAGE_LOAD.
-
-    - name: run ${{ matrix.test }} with ${{ github.event.inputs.controller-image }} image
-      if: ${{ github.event.inputs.controller-image != 'kong/kubernetes-ingress-controller:ci' }}
-      run: make test.e2e
-      env:
-        E2E_TEST_RUN: ${{ matrix.test }}
-        TEST_KONG_CONTROLLER_IMAGE_OVERRIDE: ${{ github.event.inputs.controller-image }}
-        KONG_LICENSE_DATA: ${{ steps.license.outputs.license }}
-        KONG_CLUSTER_VERSION: ${{ github.event.inputs.kubernetes-version }}
-        ISTIO_VERSION: ${{ github.event.inputs.istio-version }}
-
-    - name: run ${{ matrix.test }} with ${{ github.event.inputs.controller-image }} image (local)
-      if: ${{ github.event.inputs.controller-image == 'kong/kubernetes-ingress-controller:ci' }}
-      run: make test.e2e
-      env:
-        E2E_TEST_RUN: ${{ matrix.test }}
-        TEST_KONG_CONTROLLER_IMAGE_LOAD: ${{ github.event.inputs.controller-image }}
-        TEST_KONG_CONTROLLER_IMAGE_OVERRIDE: ${{ github.event.inputs.controller-image }}
-        TEST_KONG_KONNECT_ACCESS_TOKEN: ${{ secrets.K8S_TEAM_KONNECT_ACCESS_TOKEN }}
-        KONG_LICENSE_DATA: ${{ steps.license.outputs.license }}
-        KONG_CLUSTER_VERSION: ${{ github.event.inputs.kubernetes-version }}
-        ISTIO_VERSION: ${{ github.event.inputs.istio-version }}
-
-    - name: upload diagnostics
-      if: ${{ always() }}
-      uses: actions/upload-artifact@v3
-      with:
-        name: diagnostics-e2e-tests
-        path: /tmp/ktf-diag*
-        if-no-files-found: ignore
-
-  integration-tests:
-    if: ${{ github.event.inputs.include-integration == 'true' }}
-    runs-on: ubuntu-latest
+    needs: build-image
```

Comment:
I wonder how does this work if `build-image` is conditional? I'd assume that when `build-image` doesn't run because of the condition that it has defined this works because the branch which doesn't need its outputs is chosen. Correct?

Reply:
👍